### PR TITLE
[22463] Fix Discovery CLI Tool in Windows (No privileges)

### PR DESCRIPTION
**Pull request by cferreiragonz**
## Description

When Fast DDS is compiled without admin privileges, no symlink is created to link to the `fast-discovery-server-1.0.1.exe` file. In this case, an alternative `.bat` is created to call the tool. However, the `.bat` file contains an error and fails to properly call the tool because it adds an unknown parameter as argument. To solve this, the `.bat` file was removed and the `.exe` file is called from the Python script directly. This gives more control over the tool as it does not bypass the Python script when compiling without admin privileges. 

@Mergifyio backport 3.1.x 3.0.x

## Contributor Checklist

- [X] Commit messages follow the project guidelines. 
- [X] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- _N/A_ Any new/modified methods have been properly documented using Doxygen. 
- _N/A_ Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- [X] Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- [X] Changes are API compatible. 
- _N/A_ New feature has been added to the `versions.md` file (if applicable).
- _N/A_ New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [x] Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] If this is a critical bug fix, backports to the critical-only supported branches have been requested.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Review by Mario-DL — CHANGES_REQUESTED** [review]
Nice addition, I think getting rid of the `.bat` file makes sense.
Changes LGTM and tested in my local.

We would need to add `psutil` to the `windows` and `mac` reusable workflows just at the point where we specify the required Python dependencies `vcstool` `xmlschema`

File: tools/fastdds/discovery/parser.py [hunk 1]
```diff
@@ -94,9 +94,14 @@ def __find_tool_path(self):
...
             if not os.path.exists(ret):
-                ret = tool_path / 'fast-discovery-server.bat'
-                if not os.path.exists(ret):
-                    print('fast-discovery-server tool not installed')
+                exe_files = [f for f in tool_path.glob('*.exe') if re.match(r'fast-discovery-server.*\.exe$', f.name)]
+                if len(exe_files) == 0:
+                    print("Unable to find fast-discovery-server tool. Check installation")
+                elif len(exe_files) == 1:
+                    ret = exe_files[0]
+                    print(f'[PYTHON] Found executable: {ret}')
```
[the remark this answers is not in the document]
Comment (cferreiragonz) [reply tools/fastdds/discovery/parser.py:102]: Nice catch!

**Review by Mario-DL — APPROVED** [review]
LGTM with green CI

**Comment by cferreiragonz** [comment]
@Mergifyio backport 3.1.x 3.0.x

**Pull request merged**